migrations/2023-06-16-220750_create_controversial_indexes/up.sql

```diff
@@ -0,0 +1,33 @@
+-- Need to add immutable to the controversy_rank function in order to index by it
+
+-- Controversy Rank:
+--      if downvotes <= 0 or upvotes <= 0:
+--          0 
+--      else:
+--          (upvotes + downvotes) * min(upvotes, downvotes) / max(upvotes, downvotes) 
+create or replace function controversy_rank(upvotes numeric, downvotes numeric, score numeric)
+returns integer as $$
+begin
+    if downvotes <= 0 or upvotes <= 0 then
+        return 0;
+    else
+        return floor((upvotes + downvotes) * case when upvotes > downvotes then downvotes::float / upvotes::float else upvotes::float / downvotes::float end);
+    end if;
```

Comment:
One thing that came up in #2515 was what the sorting algorithm looks like. [The algorithm used by Reddit (in 2015)](https://github.com/reddit-archive/reddit/blob/753b17407e9a9dca09558526805922de24133d53/r2/r2/lib/db/_sorts.pyx#L60) looks to me like it would work better. Comparison here: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=400c37593a339dcdb53851e685ecaf75

Note that I changed this from the one in the issue to use a float for the Reddit one, which I think works better for controversy.